Zilliqa/contracts/compressed/minimize_proxy.scilla

```diff
@@ -35,14 +36,41 @@ in
 
 contract ProxyContract
 (
+contract_owner: ByStr20,
+name :    String,
+symbol :  String,
+decimals : Uint32,
+init_supply : Uint128,
 init_implementation : ByStr20,
 init_admin : ByStr20
 )
+with
+let string_is_not_empty =
+fun (s : String) =>
+let zero = Uint32 0 in
+let s_length = builtin strlen s in
+let s_empty = builtin eq s_length zero in
+negb s_empty
+in
+let name_ok = string_is_not_empty name in
+let symbol_ok = string_is_not_empty symbol in
+let name_symbol_ok = andb name_ok symbol_ok in
+let decimals_ok =
+let six = Uint32 6 in
+let eighteen = Uint32 18 in
+let decimals_at_least_6 = uint32_le six decimals in
+let decimals_no_more_than_18 = uint32_le decimals eighteen in
+andb decimals_at_least_6 decimals_no_more_than_18 in
+andb name_symbol_ok decimals_ok
+=>
 
 field implementation : ByStr20 = init_implementation
 field admin : ByStr20 = init_admin
-field balances : Map ByStr20 Uint128 = Emp ByStr20 Uint128
-field totalSupply : Uint128 = zero
```

Comment:
seems this could fail current middleware total_supply API 
so let me know if we deployed new contract, and I'll cover middleware upgrades

<img width="520" alt="截圖 2020-07-21 上午11 31 31" src="https://user-images.githubusercontent.com/53717774/88009722-bc445c00-cb45-11ea-8aa8-b731b3646ca8.png">

Reply:
Yes you are right, good catch! Changing allowed -> allowances doesn't affect hor?

Reply:
We have an endpoint calling transferFrom but

1) it's called `SendFromHdWallet` which I assume Weiyi created this initially because we don't have ONC integrated yet and it uses the user's ID as an index into address generation from the mnemonic. 
2) the transferFrom code does not check allowances, it simply creates the transaction template, signs it and broadcasts it. 

I think this is not an issue but let's wait for @ChiuWeiYi to confirm!

Reply:
correct, it's not an issue with current states!

Reply:
can get back to here once we taking native transfer instead of ONC